packages/nextjs/components/scaffold-eth/Faucet.tsx

```diff
@@ -28,7 +27,7 @@ export const Faucet = () => {
   useEffect(() => {
```

Comment:
This whole `useEffect` feels a bit off.

Maybe this can be cleaned up like so:
```typescript
useEffect(() => {
  provider
    ?.listAccounts()
    .then(accounts => setFaucetAddress(accounts[FAUCET_ACCOUNT_INDEX]))
    .catch(error => {
      notification.error(
        <>
          <p className="font-bold mt-0 mb-1gi">Cannot connect to local provider</p>
          <p className="m-0">Did you forget to run `yarn chain`?</p>
        </>,
      );
      console.error("⚡️ ~ file: Faucet.tsx:getFaucetAddress ~ error", error);
    });
}, [provider]);
```

I also don't understand why the JSON stringify is needed. Feels wrong, and there's probably a cleaner way to fix the issue that it might cause.

Reply:
> I also don't understand why the JSON stringify is needed. Feels wrong, and there's probably a cleaner way to fix the issue that it might cause.

Yeah I agree but since the provider is an `object` as a dependency and react uses referential diff for deps array which causes `useEffect` to run multiple time : 

https://user-images.githubusercontent.com/80153681/233550082-aee7cfa0-be73-414d-9e09-b7a21fb344eb.mov

I think `JSON.stringigy` is one of the used options to do deep equals

 Nevertheless, I memoized the provider since `localhost` is static by wrapping  `getLocalProvider` with `useMemo` and it does the job  [eb3a0de](https://github.com/scaffold-eth/scaffold-eth-2/pull/292/commits/eb3a0dec3518affa4036953cd5da1b714598872e) thanks 🙌

Reply:
> Maybe this can be cleaned up like so:

Not strong on this, but looks minimal and nice, I think it's worth mentioning in #232 which pattern are we following `then/catch` vs `async/await` 🙌

Reply:
I agree that it looks a bit better, but I personally prefer to follow the async/await pattern like in the rest of the project.

Reply:
Sure, no issues with keeping things consistent :+1: